src/engine/RaceManager.cpp

```diff
@@ -13,8 +13,45 @@ extern "C" {
 RaceManager::RaceManager(World& world) : WorldContext(world) {
 }
 
+std::unordered_map<uintptr_t, std::shared_ptr<Vtx>> mirroredVtxCache;
+
+extern "C" void add_triangle_to_collision_mesh(Vtx* vtx1, Vtx* vtx2, Vtx* vtx3, Vtx** outVtx1, Vtx** outVtx2, Vtx** outVtx3) {
+    if (gIsMirrorMode != 0) {
+        auto getOrCreateMirrored = [](Vtx* original) -> Vtx* {
+            uintptr_t key = reinterpret_cast<uintptr_t>(original);
+            
+            auto it = mirroredVtxCache.find(key);
+            if (it != mirroredVtxCache.end()) {
+                return it->second.get();
+            }
+
+            auto newVtx = std::make_shared<Vtx>(*original);
+            newVtx->v.ob[0] = -newVtx->v.ob[0];
+
+            mirroredVtxCache[key] = newVtx;
+
+            return newVtx.get();
+        };
```

Comment:
We're going to need to discuss this function location a little bit, lol.

You can actually run any model through the following functions and it will add it to collision:
`generate_collision_mesh_with_defaults()`
`generate_collision_mesh_with_default_section_id`
`generate_collision_mesh()`

You can see an example of this in MarioRaceway.cpp as that's how it adds the green pipe to the track.

`add_triangle_to_collision_mesh` this function is fine too. Maybe in Game.cpp since that's generally where random API functions get spammed?